src/cpp/session/modules/SessionPatches.R

```diff
@@ -13,24 +13,42 @@
 #
 #
 
-setHook(
-   packageEvent("rstudioapi", "onLoad"),
-   function(...) {
+# NOTE: registered hooks will be run immediately if the
+# package has already been loaded.
+.rs.addFunction("registerPackageLoadHook", function(package, hook)
+{
+   # if the package is already loaded, run the hook;
+   # otherwise, register a load hook
+   if (package %in% loadedNamespaces())
+      hook()
+   else
+      setHook(packageEvent(package, "onLoad"), hook)
+})
 
-      # bail if we're not version 0.7
-      if (packageVersion("rstudioapi") != "0.7")
-         return()
+.rs.registerPackageLoadHook("rstudioapi", function(...)
+{
+   # bail if we're not version 0.7
+   if (packageVersion("rstudioapi") != "0.7")
+      return()
+   
+   # re-assign the 'selectDirectory()' function
+   rstudioapi <- asNamespace("rstudioapi")
+   override <- function(caption = "Select Directory",
+                        label = "Select",
+                        path = rstudioapi::getActiveProject())
+   {
+      callFun("selectDirectory", caption, label, path)
+   }
+   environment(override) <- rstudioapi
+   
+   .rs.replaceBinding("selectDirectory", "rstudioapi", override)
+})
 
-      # re-assign the 'selectDirectory()' function
-      rstudioapi <- asNamespace("rstudioapi")
-      override <- function(caption = "Select Directory",
-                           label = "Select",
-                           path = rstudioapi::getActiveProject())
-      {
-         callFun("selectDirectory", caption, label, path)
-      }
-      environment(override) <- rstudioapi
+.rs.registerPackageLoadHook("parallel", function(...)
+{
+   # enforce sequential setup of cluster on macOS
+   # https://github.com/rstudio/rstudio/issues/6692
+   if (Sys.info()[["sysname"]] == "Darwin")
+      parallel:::setDefaultClusterOptions(setup_strategy = "sequential")
```

Comment:
Should we only do this if R >= 4.0?

Reply:
`setDefaultClusterOptions()` does accept "arbitrary" options, even those not known by a particular version of `parallel`, so this should be safe.